network/p2p/inspector/validation/control_message_validation_inspector.go

```diff
@@ -810,34 +926,22 @@ func (c *ControlMsgValidationInspector) checkClusterPrefixHardThreshold(nodeID f
 // Args:
 //   - req: inspect rpc request that failed validation.
 //   - ctlMsgType: the control message type of the rpc message that caused the error.
-//   - err: the error that occurred.
-//   - count: the number of occurrences of the error.
-//   - isClusterPrefixed: indicates if the errors occurred on a cluster prefixed topic.
-func (c *ControlMsgValidationInspector) logAndDistributeAsyncInspectErrs(req *InspectRPCRequest, ctlMsgType p2pmsg.ControlMessageType, err error, count uint64, topicType p2p.CtrlMsgTopicType) {
+//   - errs: the errors that occurred.
+func (c *ControlMsgValidationInspector) logAndDistributeAsyncInspectErrs(req *InspectRPCRequest, ctlMsgType p2pmsg.ControlMessageType, errs p2p.InvCtrlMsgErrs) {
 	lg := c.logger.With().
-		Err(err).
+		Err(errs.Error()).
 		Str("control_message_type", ctlMsgType.String()).
 		Bool(logging.KeySuspicious, true).
 		Bool(logging.KeyNetworkingSecurity, true).
-		Str("topic_type", topicType.String()).
-		Uint64("error_count", count).
+		Int("error_count", errs.Len()).
 		Str("peer_id", p2plogging.PeerId(req.Peer)).
 		Logger()
-
-	switch {
-	case IsErrActiveClusterIDsNotSet(err):
-		lg.Warn().Msg("active cluster ids not set")
-	case IsErrUnstakedPeer(err):
-		lg.Warn().Msg("control message received from unstaked peer")
-	default:
-		distErr := c.distributor.Distribute(p2p.NewInvalidControlMessageNotification(req.Peer, ctlMsgType, err, count, topicType))
-		if distErr != nil {
-			lg.Error().
-				Err(distErr).
-				Msg("failed to distribute invalid control message notification")
-		}
-		lg.Error().Msg("rpc control message async inspection failed")
+	err := c.distributor.Distribute(p2p.NewInvalidControlMessageNotification(req.Peer, ctlMsgType, errs))
+	if err != nil {
+		c.logAndThrowError(fmt.Errorf("failed to distribute invalid control message notification: %w", err))
 	}
+	c.metrics.InvalidControlMessageNotificationErrors(errs.Len())
```

Comment:
Suggestion: let's iterate over the `errs` and report them per topic, this helps in incident response and forensics a lot on pin-pointing the channel:

Here:
```go
	for _, err := range errs {
		c.metrics.InvalidControlMessageNotificationErrors(errs.Topic())
	}
```

in `module/metrics/gossipsub_rpc_validation_inspector.go`:
```go
        // metric definition
	gc.invalidCtrlMessageErrorCount = promauto.NewCounterVec(
		prometheus.CounterOpts{
			Namespace: namespaceNetwork,
			Subsystem: subsystemGossip,
			Name:      gc.prefix + "rpc_invalid_control_message_error_total",
			Help:      "total number of errors in a single invalid control message notification",
		}, []string{LabelChannel},
	)
	
        // metric collector event handler
	func (c *GossipSubRpcValidationInspectorMetrics) InvalidControlMessageNotificationError(topic string) {
	     c.invalidCtrlMessageErrorCount.WithLabelValues(topic).Inc()
        }

```

Reply:
https://github.com/onflow/flow-go/pull/4978/commits/77530d00fece3ba71101a7898a0ae885e5f6caea

Also created followup issue to refactor and improve metrics visibility of RPC inspector https://github.com/dapperlabs/flow-internal/issues/1905